src/OpenTelemetry.Extensions.Hosting/Implementation/SelfDiagnosticsEventLogForwarder.cs

```diff
@@ -0,0 +1,158 @@
+            {
+                this.EnableEvents(eventSource, this.logLevel, EventKeywords.All);
+            }
+        }
+
+        protected override void OnEventSourceCreated(EventSource eventSource)
+        {
+            if (eventSource.Name.StartsWith(EventSourceNamePrefix, StringComparison.Ordinal))
+            {
+                // If there are EventSource classes already initialized as of now, this method would be called from
+                // the base class constructor before the first line of code in SelfDiagnosticsEventListener constructor.
+                // In this case logLevel is always its default value, "LogAlways".
+                // Thus we should save the event source and enable them later, when code runs in constructor.
+                if (this.eventSourcesBeforeConstructor != null)
+                {
+                    lock (this.lockObj)
+                    {
+                        if (this.eventSourcesBeforeConstructor != null)
+                        {
+                            this.eventSourcesBeforeConstructor.Add(eventSource);
+                            return;
+                        }
+                    }
+                }
+
+                this.EnableEvents(eventSource, this.logLevel, EventKeywords.All);
+            }
+
+            base.OnEventSourceCreated(eventSource);
+        }
+
+        /// <summary>
+        /// This method records the events from event sources to the logging system.
+        /// </summary>
+        /// <param name="eventData">Data of the EventSource event.</param>
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            if (this.loggerFactory == null)
+            {
+                return;
+            }
+
+            var logger = this.loggers.GetOrAdd(eventData.EventSource.Name, name => this.loggerFactory.CreateLogger(ToLoggerName(name)));
+            logger.Log(MapLevel(eventData.Level), new EventId(eventData.EventId, eventData.EventName), new EventSourceEvent(eventData), null, this.formatMessage);
+        }
+
+        private static string ToLoggerName(string name)
+        {
+            return name.Replace('-', '.');
+        }
+
+        private static LogLevel MapLevel(EventLevel level)
+        {
+            switch (level)
+            {
+                case EventLevel.Critical:
+                    return LogLevel.Critical;
+                case EventLevel.Error:
+                    return LogLevel.Error;
+                case EventLevel.Informational:
+                    return LogLevel.Information;
+                case EventLevel.Verbose:
+                    return LogLevel.Debug;
+                case EventLevel.Warning:
+                    return LogLevel.Warning;
+                case EventLevel.LogAlways:
+                    return LogLevel.Information;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(level), level, null);
```

Comment:
Throw or swallow?

Reply:
Removed throw and just set to LogLevel.None.